src/session/components/message_entry.rs

```diff
@@ -117,24 +146,36 @@ impl MessageEntry {
         let imp = self.imp();
         let buffer = imp.text_view.buffer();
         let text = buffer
```

Comment:
Can't make a proper suggestion here as there are deleted lines in between, but I'm wondering whether it's better to use `Into` here. It seems there is a chance that the underlying string is moved instead of being copied.

```suggestion
        let text: String = buffer
            .text(&buffer.start_iter(), &buffer.end_iter(), true)
            .into();
```